spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/properties/KafkaBinderConfigurationProperties.java

```diff
@@ -98,6 +99,11 @@
 
 	private JaasLoginModuleConfiguration jaas;
 
+	/**
+	 * The bean name of a custom header mapper to use instead of a {@link DefaultKafkaHeaderMapper}.
+	 */
+	private String headerMapperBeanName;
```

Comment:
Why can't we rely on the dependency injection?
I mean we provide the default `DefaultKafkaHeaderMapper` bean in the `KafkaBinderConfiguration` with the `@ConditionalOnMissingBean` and, therefore, if end-user provides its own bean on the matter, we will get exactly that injected.

At least that is how the `ProducerListener` is going to work. And that is how I do in the Kinesis Binder for the `MetadataStore`: https://github.com/spring-cloud/spring-cloud-stream-binder-aws-kinesis/blob/master/spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/config/KinesisBinderConfiguration.java#L83

WDYT?

Reply:
That won't work because `headerPatterns` can be specified at the binding level, so we need an instance per binding.

Also, the user might want a different mapper for each binding.